src/kixi/heimdall/handler.clj

```diff
@@ -54,6 +61,22 @@
       {:status 201 :body res}
       {:status 401 :body res})))
 
+(defn escape-html
+  "Change special characters into HTML character entities."
+  [text]
+  (.. #^String (str text)
+      (replace "&" "&amp;")
+      (replace "<" "&lt;")
+      (replace ">" "&gt;")
+      (replace "\"" "&quot;")))
+
+(defn wrap-escape-html
+  [handler]
+  (fn [request]
+    (handler (update-in request [:params] #(clojure.walk/prewalk (fn [v] (if (string? v)
+                                                                           (escape-html v)
+                                                                           v)) %)))))
```

Comment:
again, I'm a bit surprised that there isn't something from @weavejester or someone else that does this for us.